File: 11-ContainerWithMostWater/cpp/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_area = 0;
        int i = 0;
        int j = height.size()-1;
        while(i<j){
            int h = min(height[i], height[j]);
            max_area = max(max_area, ((j-i)*h));
            if(height[i]<height[j]){
                i++;
            }
            else{
                j--;
            }
        }
        return max_area;
    }
};

File: 11-ContainerWithMostWater/javascript/solution.js
// Last updated: 3/6/2026, 12:46:28 AM
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let max = 0;
    let i = 0;
    let j = height.length - 1;
    while(i < j){
        let min = Math.min(height[i], height[j]);
        max = Math.max(max, min * (j - i));
        if(height[i] < height[j]){
            i++;
        }
        else{
            j--;
        }
    }
    return max
};